### PR TITLE
static pages

### PR DESCRIPTION
**Pull request by MichalPlachyFramgia**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,45 +1,18 @@
 source 'https://rubygems.org'
 
-
-# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '4.2.1'
-# Use sqlite3 as the database for Active Record
 gem 'sqlite3'
-# Use SCSS for stylesheets
+gem 'bootstrap-sass', '~> 3.3.5'
```
Comment (dieunb) [inline Gemfile]: replace all `'` with `"`

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_locale
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: ???. I don't understand what do you want?

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
-end
+
+  def full_title page_title = ''
```
Comment (dieunb) [inline app/helpers/application_helper.rb]: use `"`

File: app/views/static_pages/about.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<% provide(:title, t('about')) %>
```
Comment (dieunb) [inline app/views/static_pages/about.html.erb]: remove `()`

File: app/views/static_pages/contact.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<% provide(:title, t('contact')) %>
```
Comment (dieunb) [inline app/views/static_pages/contact.html.erb]: remove `()`

File: app/views/static_pages/help.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<% provide(:title, t('help')) %>
```
Comment (dieunb) [inline app/views/static_pages/help.html.erb]: remove `()`

File: config/routes.rb [hunk 7]
```diff
@@ -1,56 +1,7 @@
...
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  root                'static_pages#home'
+  get    'help'    => 'static_pages#help'
+  get    'about'   => 'static_pages#about'
+  get    'contact' => 'static_pages#contact'
+  get    'signup'  => 'users#new'
```
Comment (dieunb) [inline config/routes.rb]: ???. I don't see users_controller

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (MichalPlachyFramgia) [inline app/controllers/application_controller.rb]: Just for i18n

File: config/routes.rb (hunk 7 above)
Comment (MichalPlachyFramgia) [inline config/routes.rb]: added users_controller

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
-end
+
+  def full_title page_title = ""
+    base_title = "Framgia E-Learning System"
```
Comment (phamkykhoi) [inline app/helpers/application_helper.rb]: I18N

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t('home'), root_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_header.html.erb [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+        <li><%= link_to t('help'), help_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_header.html.erb [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+        <li><%= link_to t('contact'), contact_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_header.html.erb]: I18n

File: config/routes.rb [hunk 12]
```diff
@@ -1,56 +1,7 @@
...
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  root                'static_pages#home'
```
Comment (phamkykhoi) [inline config/routes.rb]: ' ' ==> " "

File: config/routes.rb [hunk 13]
```diff
@@ -1,56 +1,7 @@
...
+  get    'help'    => 'static_pages#help'
```
Comment (phamkykhoi) [inline config/routes.rb]: ' ' ==> " "

File: config/routes.rb (hunk 7 above)
Comment (dieunb) [inline config/routes.rb]: That 's not right! Your name pull request is `static pages` and i think you should't push any thing, that is Irrelevant.

File: app/controllers/users_controller.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: Your pull request is `static pages`, i think this file is unnecessary.

File: app/controllers/application_controller.rb [hunk 15]
```diff
@@ -2,4 +2,5 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
```
Comment (nguyenduyta) [inline app/controllers/application_controller.rb]: remove this line blank

File: app/views/static_pages/home.html.erb [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t('home') %></h1>
+<div class="center jumbotron">
+  <h1>Welcome to the Sample App</h1>
```
Comment (nguyenduyta) [inline app/views/static_pages/home.html.erb]: I18

File: config/routes.rb [hunk 17]
```diff
@@ -1,56 +1,6 @@
...
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  root                "static_pages#home"
+  get    "help"    => "static_pages#help"
```
Comment (nguyenduyta) [inline config/routes.rb]: 1 space instead of more space.

File: app/helpers/application_helper.rb [hunk 18]
```diff
@@ -1,2 +1,11 @@
...
+
+  def full_title page_title = ""
+    base_title = t (:elearning)
+    if page_title.empty?
+      base_title
+    else
+      page_title + " | " + base_title
+    end
+  end
+end
```
Comment (khanhkhanh) [inline app/helpers/application_helper.rb]: Break line

File: app/helpers/application_helper.rb [hunk 19]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
-end
+
+  def full_title page_title = ""
+    base_title = t (:elearning)
```
Comment (khanhkhanh) [inline app/helpers/application_helper.rb]: Remove ()

File: app/views/layouts/application.html.erb [hunk 20]
```diff
@@ -1,14 +1,22 @@
...
+    <div class="container">
+      <% flash.each do |message_type, message| %>
+        <div class="alert alert-<%= message_type %>"><%= message %></div>
+      <% end %>
+      <%= yield %>
+      <%= render "layouts/footer" %>
+      <%= debug(params) if Rails.env.development? %>
+    </div>
+  </body>
+</html>
```
Comment (khanhkhanh) [inline app/views/layouts/application.html.erb]: Break line

File: app/views/static_pages/about.html.erb [hunk 21]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("about") %>
+<h1><%= t("about") %></h1>
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
+  </h2>
+</div>
```
Comment (khanhkhanh) [inline app/views/static_pages/about.html.erb]: Break line

File: app/views/static_pages/contact.html.erb [hunk 22]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("contact") %>
+<h1><%= t("contact") %></h1>
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
+  </h2>
+</div>
```
Comment (khanhkhanh) [inline app/views/static_pages/contact.html.erb]: Break line

File: app/views/static_pages/help.html.erb [hunk 23]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("help") %>
+<h1><%= t("help") %></h1>
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
+  </h2>
+</div>
```
Comment (khanhkhanh) [inline app/views/static_pages/help.html.erb]: Break line

File: app/views/static_pages/home.html.erb [hunk 24]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t("home") %></h1>
+<div class="center jumbotron">
+  <h1><%= t("welcome") %></h1>
+  <h2>
+    <%= t("home-description") %>
+  </h2>
+  <%= link_to t("signup-button"), "#", class: "btn btn-lg btn-primary" %>
+</div>
```
Comment (khanhkhanh) [inline app/views/static_pages/home.html.erb]: Break line

File: app/views/static_pages/about.html.erb [hunk 25]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("about") %>
+<h1><%= t("about") %></h1>
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
```
Comment (khanhkhanh) [inline app/views/static_pages/about.html.erb]: Remove ()

File: app/views/static_pages/about.html.erb [hunk 26]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("about") %>
+<h1><%= t("about") %></h1>
```
Comment (khanhkhanh) [inline app/views/static_pages/about.html.erb]: Remove ()

File: app/views/static_pages/contact.html.erb [hunk 27]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("contact") %>
+<h1><%= t("contact") %></h1>
```
Comment (khanhkhanh) [inline app/views/static_pages/contact.html.erb]: Remove ()

File: app/views/static_pages/contact.html.erb [hunk 28]
```diff
@@ -0,0 +1,7 @@
...
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
```
Comment (khanhkhanh) [inline app/views/static_pages/contact.html.erb]: Remove ()

File: app/views/static_pages/help.html.erb [hunk 29]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("help") %>
+<h1><%= t("help") %></h1>
+<div class="center jumbotron">
+  <h2>
+    <%= t("help-description") %>
```
Comment (khanhkhanh) [inline app/views/static_pages/help.html.erb]: Remove ()

File: app/views/static_pages/help.html.erb [hunk 30]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("help") %>
+<h1><%= t("help") %></h1>
```
Comment (khanhkhanh) [inline app/views/static_pages/help.html.erb]: Remove ()

File: app/views/static_pages/home.html.erb [hunk 31]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t("home") %></h1>
+<div class="center jumbotron">
+  <h1><%= t("welcome") %></h1>
```
Comment (khanhkhanh) [inline app/views/static_pages/home.html.erb]: Remove ()

File: app/views/static_pages/home.html.erb [hunk 32]
```diff
@@ -0,0 +1,8 @@
...
+  <h2>
+    <%= t("home-description") %>
```
Comment (khanhkhanh) [inline app/views/static_pages/home.html.erb]: Remove ()

File: app/views/layouts/_footer.html.erb [hunk 33]
```diff
@@ -0,0 +1,10 @@
+<footer class="footer">
+  <small>
+     <%= link_to t("elearning"), root_path %>
```
Comment (phamkykhoi) [inline app/views/layouts/_footer.html.erb]: Remove ( )

File: app/views/layouts/_footer.html.erb [hunk 34]
```diff
@@ -0,0 +1,10 @@
...
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t("about"), about_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_footer.html.erb]: Remove ()

File: app/views/layouts/_header.html.erb [hunk 35]
```diff
@@ -0,0 +1,11 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t("home"), root_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_header.html.erb]: Remove ()

File: app/views/layouts/_header.html.erb [hunk 36]
```diff
@@ -0,0 +1,11 @@
...
+        <li><%= link_to t("help"), help_path %></li>
```
Comment (phamkykhoi) [inline app/views/layouts/_header.html.erb]: Remove ( )

File: app/views/layouts/application.html.erb [hunk 37]
```diff
@@ -1,14 +1,22 @@
...
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title(yield(:title)) %></title>
+    <%= stylesheet_link_tag "application", media: "all",
+                                           "data-turbolinks-track" => true %>
```
Comment (phamkykhoi) [inline app/views/layouts/application.html.erb]: not use => 
use :

File: app/views/layouts/application.html.erb [hunk 38]
```diff
@@ -1,14 +1,22 @@
...
+    <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
```
Comment (phamkykhoi) [inline app/views/layouts/application.html.erb]: not use =>
use :

File: app/views/layouts/_footer.html.erb [hunk 39]
```diff
@@ -0,0 +1,10 @@
+<footer class="footer">
+  <small>
+     <%= link_to t('elearning'), root_path %>
```
Comment (khanhkhanh) [inline app/views/layouts/_footer.html.erb]: ' => "

File: app/views/static_pages/home.html.erb [hunk 40]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t("home") %></h1>
+<div class="center jumbotron">
+  <h1><%= t "welcome" %></h1>
+  <h2>
+    <%= t "home-description" %>
+  </h2>
+  <%= link_to t("signup-button"), "#", class: "btn btn-lg btn-primary" %>
+</div>
```
Comment (khanhkhanh) [inline app/views/static_pages/home.html.erb]: Break line

File: config/locales/vi.yml [hunk 41]
```diff
@@ -0,0 +1,15 @@
...
+  home: "Trang chủ"
+  help: "Trợ giúp"
+  signup: "Đăng ký"
+  here: "tại đây"
+  home-description: "This is the home page for the Framgia E-Learning System application."
+  help-description: "This is the help page for the Framgia E-Learning System application.
+    This is the help page for the Framgia E-Learning System application.
+    This is the help page for the Framgia E-Learning System application.
+    This is the help page for the Framgia E-Learning System application.
+    This is the help page for the Framgia E-Learning System application."
```
Comment (khanhkhanh) [inline config/locales/vi.yml]: Break line

File: app/assets/stylesheets/custom.css.scss [hunk 42]
```diff
@@ -0,0 +1,227 @@
...
+
+.users {
+  list-style: none;
+  margin: 0;
+  li {
+    overflow: auto;
+    padding: 10px 0;
+    border-bottom: 1px solid $gray-lighter;
+  }
+}
```
Comment (khanhkhanh) [inline app/assets/stylesheets/custom.css.scss]: Break line

File: config/locales/vi.yml (hunk 41 above)
Comment (phamkykhoi) [inline config/locales/vi.yml]: Not use languge vietname

**Comment by phamkykhoi** [comment]
OK

**Pull request merged**